portmanteau-binary/src/Portmanteau/Binary/Codec.hs

```diff
@@ -0,0 +1,53 @@
+{-# LANGUAGE FlexibleInstances #-}
+{-# LANGUAGE NoImplicitPrelude #-}
+{-# OPTIONS_GHC -fno-warn-orphans #-}
+module Portmanteau.Binary.Codec (
+    BinaryCodec'
+  , BinaryCodec
+  , encode
+  , decodeOrFail
+  , binaryCodec
+  , binaryCodec'
+  ) where
+
+import           Data.Binary (Binary, Get, Put)
+import           Data.Binary.Get (ByteOffset)
+import qualified Data.Binary as B
+import qualified Data.Binary.Get as B
+import qualified Data.Binary.Put as B
+import qualified Data.ByteString.Lazy as BSL
+import           Data.Functor.Contravariant (Op (..))
+
+import           P
+
+import           Portmanteau.Core
+
+
+type BinaryCodec' = Codec' (Op Put) Get
+type BinaryCodec a = BinaryCodec' a a
+
+
+-- Forgive me. I have to wonder why they haven't implemented this?
+instance Monoid Put where
```

Comment:
Yeah we should newtype it if we really want a `Monoid` instance. It's annoying not having it, but it's also not a huge deal, you can just use `>>` instead of `<>`.